### PR TITLE
refactor: remove unnecessary JsInfo hasattr calls

### PR DESCRIPTION
**Pull request by jbedard**
### Test plan

- Covered by existing test cases

### CODE REVIEW

File: ts/private/ts_project.bzl [hunk 1]
```diff
@@ -19,12 +19,12 @@ def _gather_types_from_js_infos(targets):
     files_depsets.extend([
         target[JsInfo].transitive_types
         for target in targets
-        if JsInfo in target and hasattr(target[JsInfo], "transitive_types")
```
Comment (thesayyn) [inline ts/private/ts_project.bzl:22]: Hmm, this looks like it was put here for backwards compatibility. Are we sure that it no longer an issue?

Comment (jbedard) [reply ts/private/ts_project.bzl:22]: rules_ts v3 (which `main` is for) only supports rules_js 2+ (and bazel-lib 2+) so I think this is ok

**Review by thesayyn — APPROVED** [review]
LGTM

File: ts/private/ts_project.bzl (hunk 1 above)
Comment (thesayyn) [reply ts/private/ts_project.bzl:22]: okay, that makes sense

**Pull request merged**